cpp/src/lists/contains.cu

```diff
@@ -237,5 +362,57 @@ std::unique_ptr<column> contains(cudf::lists_column_view const& lists,
   return detail::contains(lists, search_keys, rmm::cuda_stream_default, mr);
 }
 
+std::unique_ptr<column> contains_null_elements(cudf::lists_column_view const& input_lists,
+                                               rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+
+  auto stream = rmm::cuda_stream_default;
+
+  auto const num_rows   = input_lists.size();
+  auto const d_lists    = column_device_view::create(input_lists.parent());
+  auto has_nulls_output = make_numeric_column(
+    data_type{type_id::BOOL8}, input_lists.size(), mask_state::UNALLOCATED, stream, mr);
+  auto const output_begin = has_nulls_output->mutable_view().begin<bool>();
+  thrust::tabulate(
+    rmm::exec_policy(stream),
+    output_begin,
+    output_begin + num_rows,
+    [lists = cudf::detail::lists_column_device_view{*d_lists}] __device__(auto list_idx) {
+      auto list       = list_device_view{lists, list_idx};
+      auto list_begin = thrust::make_counting_iterator(size_type{0});
+      return list.is_null() ||
+             thrust::any_of(thrust::seq, list_begin, list_begin + list.size(), [&list](auto i) {
+               return list.is_null(i);
+             });
+    });
+  auto const validity_begin = cudf::detail::make_counting_transform_iterator(
+    0, [lists = cudf::detail::lists_column_device_view{*d_lists}] __device__(auto list_idx) {
+      return not list_device_view{lists, list_idx}.is_null();
+    });
+  auto [null_mask, num_nulls] = cudf::detail::valid_if(
+    validity_begin, validity_begin + num_rows, thrust::identity<bool>{}, stream, mr);
+  has_nulls_output->set_null_mask(std::move(null_mask), num_nulls);
+  return has_nulls_output;
```

Comment:
There's a lot of code here using the default stream. Should probably be put in a `detail` function that takes a stream which is called from here and passed the default stream.

Reply:
This is the inefficient implementation, to be replaced when #9588 is available. I figured I'd introduce one temporary function instead of two.

Reply:
Added `detail` API.